SmartDeviceLink_Example/SubscribeButtonManager.m

```diff
@@ -29,69 +28,41 @@ - (instancetype)initWithManager:(SDLManager *)manager {
     }
 
     _sdlManager = manager;
-    _presetButtonSubscriptionIDs = [NSMutableDictionary dictionary];
 
     return self;
 }
 
-- (void)subscribeToPresetButtons {
+- (void)subscribeToAllPresetButtons {
     if (self.sdlManager.systemCapabilityManager.defaultMainWindowCapability.numCustomPresetsAvailable.intValue == 0) {
         SDLLogW(@"The module does not support preset buttons");
         return;
     }
 
-    for (SDLButtonName buttonName in [self.class presetButtons]) {
-        if (self.presetButtonSubscriptionIDs[buttonName] != nil) {
-            SDLLogW(@"The app is already subscribed to the %@ button", buttonName);
-            return;
-        }
-
-        __weak typeof(self) weakSelf = self;
-        NSObject *subscriptionID = [self.sdlManager.screenManager subscribeButton:buttonName withUpdateHandler:^(SDLOnButtonPress * _Nullable buttonPress, SDLOnButtonEvent * _Nullable buttonEvent, NSError * _Nullable error) {
-            __strong typeof(weakSelf) strongSelf = weakSelf;
-            if (error != nil || buttonPress == nil) {
-                strongSelf.presetButtonSubscriptionIDs[buttonName] = nil;
-                return;
-            }
-
-            NSString *alertMessage;
-            if ([buttonPress.buttonPressMode isEqualToEnum:SDLButtonPressModeShort]) {
-                alertMessage = [NSString stringWithFormat:@"%@ pressed", buttonPress.buttonName];
-            } else {
-                alertMessage = [NSString stringWithFormat:@"%@ long pressed", buttonPress.buttonName];
-            }
-
-            SDLAlert *alert = [AlertManager alertWithMessageAndCloseButton:alertMessage textField2:nil iconName:nil];
-            [strongSelf.sdlManager sendRPC:alert];
-        }];
-
-        self.presetButtonSubscriptionIDs[buttonName] = subscriptionID;
+    for (SDLButtonName buttonName in [self.class sdlex_allPresetButtons]) {
+        [self.sdlManager.screenManager subscribeButton:buttonName withObserver:self selector:@selector(sdlex_buttonPressEventWithButtonName:error:buttonPress:buttonEvent:)];
     }
 }
 
-- (void)unsubscribeToPresetButtons {
-    if (self.presetButtonSubscriptionIDs.count == 0) {
-        SDLLogW(@"The app is not subscribed to preset buttons");
+- (void)sdlex_buttonPressEventWithButtonName:(SDLButtonName)buttonName error:(NSError *)error buttonPress:(SDLOnButtonPress *)buttonPress buttonEvent:(SDLOnButtonEvent *)buttonEvent {
+    if (error != nil) {
```

Comment:
There appears to be no reason for the button event here. It should maybe at least log whatever is happening?

Reply:
I removed the button event param from the selector